### ISSUE TITLE
Incompatible with Flutter 3.0 (Android)

### ISSUE THREAD

**Issue by robert-virkus** [Lyokone/flutterlocation#718]
**Describe the bug**

The plugin cannot compiled for Android with Flutter 3.0. Perhaps this affects iOS, too.
Yes, I tried running `flutter clean` first.

**Expected behavior**
The plugin compiles.

**Tested on:**
 - Compile time error

**Other plugins:**
 - List of others Flutter plugins that could interfere

**Additional logs**

``` shell
e: C:\Users\User\flutter\.pub-cache\hosted\pub.dartlang.org\location-4.3.0\android\src\main\java\com\lyokone\location\FlutterLocationService.kt: (124, 1): Class 'FlutterLocationService' is not abstract and does not implement abstract member public abstract fun onRequestPermissionsResult(p0: Int, p1: Array<(out) String!>, p2: IntArray): Boolean defined in io.flutter.plugin.common.PluginRegistry.RequestPermissionsResultListener
e: C:\Users\User\flutter\.pub-cache\hosted\pub.dartlang.org\location-4.3.0\android\src\main\java\com\lyokone\location\FlutterLocationService.kt: (258, 5): 'onRequestPermissionsResult' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':location:compileReleaseKotlin'.
> Compilation error. See log for more details
```

**Comment by tsafundzic** [comment]
Workaround: https://github.com/Lyokone/flutterlocation/issues/698#issuecomment-1052832672

**Comment by rawquesh** [comment]
any update, guys?

**Comment by mitesh77** [comment]
please do update guys? as soon as

**Comment by vikilinho** [comment]
This was what I followed to fix it.

Temporary fix:
Search for the FlutterLocationService.kt and edit it
change
override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>?, grantResults: IntArray?): Boolean
to

override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray): Boolean

**Comment by SalarPro** [comment]
> This was what I followed to fix it.
> 
> Temporary fix: Search for the FlutterLocationService.kt and edit it change override fun onRequestPermissionsResult(requestCode: Int, permissions: Array?, grantResults: IntArray?): Boolean to
> 
> override fun onRequestPermissionsResult(requestCode: Int, permissions: Array, grantResults: IntArray): Boolean

override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray): Boolean {

**Comment by noahvelasco** [comment]
> This was what I followed to fix it.
> 
> Temporary fix: Search for the FlutterLocationService.kt and edit it change override fun onRequestPermissionsResult(requestCode: Int, permissions: Array?, grantResults: IntArray?): Boolean to
> 
> override fun onRequestPermissionsResult(requestCode: Int, permissions: Array, grantResults: IntArray): Boolean

whats the directory?

**Comment by SalarPro** [comment]
> > This was what I followed to fix it.
> > Temporary fix: Search for the FlutterLocationService.kt and edit it change override fun onRequestPermissionsResult(requestCode: Int, permissions: Array?, grantResults: IntArray?): Boolean to
> > override fun onRequestPermissionsResult(requestCode: Int, permissions: Array, grantResults: IntArray): Boolean
> 
> whats the directory?
mine is this
`"C:\Users\<YOUR_PC_USER>\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\location-4.3.0\android\src\main\java\com\lyokone\location\FlutterLocationService.kt"`

**Comment by zmeggyesi** [comment]
@Lyokone This sounds like a rather urgent change. Would you be able to review, merge, and release this patch?

**Comment by Lyokone** [comment]
Hey, sorry not having a lot of time to maintain the package right now, should improve soon 🤞
I'm looking at that now

**Comment by Lyokone** [comment]
It's published in 4.4.0
